packages/core/tests/anonymousSignIn.test.ts

```diff
@@ -0,0 +1,142 @@
+})
+
+test(`should fail if network is unavailable`, async () => {
+  server.use(anonymousNetworkErrorHandler)
+
+  authService.send({ type: 'SIGNIN_ANONYMOUS' })
+
+  const state: AuthState = await waitFor(authService, (state: AuthState) =>
+    state.matches('authentication.signedOut.failed')
+  )
+
+  expect(state.context.errors).toMatchInlineSnapshot(`
+    {
+      "authentication": {
+        "error": "OK",
+        "message": "Network Error",
+        "status": 200,
+      },
+    }
+  `)
+})
+
+test(`should fail if server returns an error`, async () => {
+  server.use(anonymousNetworkErrorHandler)
+
+  authService.send({ type: 'SIGNIN_ANONYMOUS' })
+
+  const state: AuthState = await waitFor(authService, (state: AuthState) =>
+    state.matches('authentication.signedOut.failed')
+  )
+
+  expect(state.context.errors).toMatchInlineSnapshot(`
+    {
+      "authentication": {
+        "error": "OK",
+        "message": "Network Error",
+        "status": 200,
+      },
+    }
+  `)
+})
+
+test('should deanonymise a user with email and password', async () => {
+  server.use(correctAnonymousHandler, deamonymisationSuccessfulHandler)
+  authService.send({ type: 'SIGNIN_ANONYMOUS' })
+
+  await waitFor(authService, (state: AuthState) => state.matches('authentication.signedIn'))
+
+  authService.send({
+    type: 'SIGNUP_EMAIL_PASSWORD',
+    email: faker.internet.email(),
+    password: faker.internet.password(15)
+  })
+
+  const state: AuthState = await waitFor(authService, (state: AuthState) =>
+    state.matches({
+      authentication: 'signedOut',
+      registration: { incomplete: 'needsEmailVerification' }
+    })
+  )
+
+  expect(state.context.user).toBeNull()
+})
+
+test('should deanonymise a user with passwordless email', async () => {
+  server.use(correctAnonymousHandler, deamonymisationSuccessfulHandler)
+  authService.send({ type: 'SIGNIN_ANONYMOUS' })
+
+  await waitFor(authService, (state: AuthState) => state.matches('authentication.signedIn'))
+
+  authService.send({
+    type: 'PASSWORDLESS_EMAIL',
+    email: faker.internet.email()
+  })
+
+  const state: AuthState = await waitFor(authService, (state: AuthState) =>
+    state.matches({
+      authentication: 'signedOut',
+      registration: { incomplete: 'needsEmailVerification' }
+    })
+  )
+
+  expect(state.context.user).toBeNull()
+})
+
+test('should deanonymise a user with passwordless sms', async () => {
+  server.use(correctAnonymousHandler, deamonymisationSuccessfulHandler)
+  authService.send({ type: 'SIGNIN_ANONYMOUS' })
+
+  await waitFor(authService, (state: AuthState) => state.matches('authentication.signedIn'))
+
+  authService.send({
+    type: 'PASSWORDLESS_SMS',
+    phoneNumber: faker.phone.phoneNumber()
+  })
+
+  const state: AuthState = await waitFor(authService, (state: AuthState) =>
+    state.matches({
+      authentication: 'signedOut',
+      registration: { incomplete: 'needsOtp' }
+    })
+  )
+
+  expect(state.context.user).toBeNull()
+})
```

Comment:
What happens if the deanonymization process is not successful? Could you please also cover that scenario with a couple of tests?